doc/developer-guides/coding_guidelines.rst

```diff
@@ -2787,3 +2722,416 @@ Compliant example::
        #define _Bool bool
 
 
+Coding Style
+************
+
+General
+=======
+
+GN-01: Each line shall contain at most 120 characters
+-----------------------------------------------------
+
+No more than 120 characters shall be on a line, with tab stops every 8
+characters. Statements longer than this limit shall be broken into multiple
+lines with proper alignment.
+
+Compliant example::
+
+    cpuid(CPUID_EXTEND_FEATURE, &unused,
+        &boot_cpu_data.cpuid_leaves[FEAT_7_0_EBX],
+        &boot_cpu_data.cpuid_leaves[FEAT_7_0_ECX],
+        &boot_cpu_data.cpuid_leaves[FEAT_7_0_EDX]);
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       cpuid(CPUID_EXTEND_FEATURE, &unused, &boot_cpu_data.cpuid_leaves[FEAT_7_0_EBX], &boot_cpu_data.cpuid_leaves[FEAT_7_0_ECX], &boot_cpu_data.cpuid_leaves[FEAT_7_0_EDX]);
+
+
+GN-02: Each line shall only contain one statement
+-------------------------------------------------
+
+Compliant example::
+
+    if (condition) {
+        do_a();
+    } else {
+        do_b();
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       if (condition) { do_a();} else {do_b();}
+
+
+GN-03: The semicolon ending a statement shall be the last character of that line
```

Comment:
Reading this again, do we need this requirement?  This would disallow having a comment such as:
```
    a = 0;   /* initialize a */
```
Are such comments not allowed?  Also, you forbid having trailing spaces in another rule, so I don't think this rule is required. Your non-compliant example here is actually covered by the previous GN-02 (only one statement per line).

Reply:
a = 0;   /* initialize a */
shall be allowed.

I will remove this rule in v4.